aviary/modules/processor.py

```diff
@@ -495,20 +499,31 @@ def process_batch(args, prefix):
 
     logging.info(f"Reading batch file: {args.batch_file}")
 
-    header=0
+    header=None
+    separator=' '
     with open(args.batch_file, mode='r') as check_batch:
         for line in check_batch.readlines():
-            if "sample\tshort_reads_1\tshort_reads_2\tlong_reads\tlong_read_type\tassembly\tcoassemble" in line \
-                or "sample,short_reads_1,short_reads_2,long_reads,long_read_type,assembly,coassemble" in line \
-                or "sample  short_reads_1   short_reads_2   long_reads      long_read_type  assembly        coassemble" in line \
-                or "sample short_reads_1 short_reads_2 long_reads long_read_type assembly coassemble" in line:
-               header=1
-               logging.debug("Inferred header")
-            else:
-                logging.debug("No heading inferred.")
+            line = line.strip()
+            for sep in ['\t', ',', ' ']:
+                separated = line.split(sep)
+                print(separated)
+                if separated == BATCH_HEADER:
+                    header=0
+                    separator=sep
+                    logging.debug("Inferred header")
+                    break
+                elif len(separated) >= 7:
+                    header=None
+                    separator=sep
+                    logging.debug("Inferred no header")
+                    break
+            if header is None:
+                logging.debug("No header found")
             break
-
-    batch = pd.read_csv(args.batch_file, sep=None, engine='python', skiprows=header)
+    if header is not None:
+        batch = pd.read_csv(args.batch_file, sep=separator, engine='python', header=header)
+    else:
+        batch = pd.read_csv(args.batch_file, sep=separator, engine='python', names=BATCH_HEADER)

```

Comment:
This part is a bit obscure to me, maybe because I'm used to the far superior polars...
So if `header is not None`, then it is set to `0` because `separated == BATCH_HEADER`, so the arg is `header=0` meaning the header is in the first line? Would you ever have it on a different line? If its not needed, then maybe a boolean would work better?

Then `else` uses just `names=BATCH_HEADER`, does this automatically assume no header?
Might be a way to just give header=TRUE/FALSE and names=BATCH_HEADER in a single line that works for both cases.

Reply:
I agree that polars is better, don't get me wrong but pandas was being used here before polars was in a stable enough state and I ain't about to change it haha.

The way header works in read_csv is kind of annoying, you can't pass `header` a boolean. And the documentation read to me like you can't pass `names` and `header` at the same time, but obviously that is wrong you can just pass it None as the default value. So yeah, this can get condensed into a single line